src/services/copyConfig/rba/rba.js

```diff
@@ -84,14 +88,39 @@ export default class Rba {
     return response
   }
 
-  async setRbaRulesAndSettings(destinationSite, destinationSiteConfiguration, payload) {
-    const response = await this.#rbaPolicy.set(destinationSite, payload, destinationSiteConfiguration.dataCenter)
+  // Merge originCommonRules into destinationCommonRules by id
+  mergeCommonRules(originCommonRules, destinationCommonRules) {
+    const destinationCommonRulesMap = new Map(destinationCommonRules.map((rule) => [rule.id, rule]))
+
+    originCommonRules.forEach((originRule) => {
+      if (destinationCommonRulesMap.has(originRule.id)) {
+        const destinationRuleIndex = destinationCommonRules.findIndex((r) => r.id === originRule.id)
+        destinationCommonRules[destinationRuleIndex] = { ...destinationCommonRulesMap.get(originRule.id), ...originRule }
+      } else {
+        destinationCommonRules.push(originRule)
+      }
+    })
+
+    return destinationCommonRules
+  }
+
+  async setRbaRulesAndSettings(destinationApiKey, destinationSiteConfiguration, payload, operation = Rba.OPERATION.MERGE) {
+    if (operation === Rba.OPERATION.MERGE) {
+      const destinationSiteRbaPolicy = new RbaPolicy(this.#credentials, destinationApiKey, destinationSiteConfiguration.dataCenter)
+      const destinationSitePolicies = await destinationSiteRbaPolicy.get()
+
+      const originCommonRules = payload.policy.commonRules
+      const destinationCommonRules = destinationSitePolicies.policy.commonRules
+      payload.policy.commonRules = this.mergeCommonRules(originCommonRules, destinationCommonRules)
+    }
+
+    const response = await this.#rbaPolicy.set(destinationApiKey, payload, destinationSiteConfiguration.dataCenter)
     response['context'] = response.context.replace(/&quot;/g, '"')
     return response
   }
 
-  async get() {
-    return await Promise.all([this.#riskAssessment.get(), this.#policies.get(), this.#rbaPolicy.get()])
+  get() {
+    return Promise.all([this.#riskAssessment.get(), this.#policies.get(), this.#rbaPolicy.get()])
```

Comment:
This change won't break the behavior? now get() do not wait anymore for the promise to be resolved with all the data to be collected.